crates/rpc/rpc-types/src/eth/block.rs

```diff
@@ -33,6 +45,80 @@ pub struct Block {
     pub base_fee_per_gas: Option<U256>,
 }
 
+impl Block {
+    /// Create a new block response from a [primitive block](reth_primitives::Block), using the
+    /// total difficulty to populate its field in the rpc response.
+    pub fn from_block_full(
+        block: PrimitiveBlock,
```

Comment:
Maybe stupid Q, why do we have 2 block types here? How can you unify them?

Reply:
Also the signed transactions in the `Block` type do not have a sender field, whereas rpc clients expect this in a `from` field